src/components/Button/Button.tsx

```diff
@@ -1,12 +1,45 @@
 import * as React from 'react';
+import { oneOf } from "prop-types";
 import {Button as CoreButton, ButtonProps as CoreButtonProps} from 'wix-ui-core/Button';
-import style from './Button.st.css';
 import {withStylable} from 'wix-ui-core/withStylable';
+import { enumValues } from '../../utils';
+import style from './Button.st.css';
+import {Skin, Priority, Size} from './constants';
+
+export interface ButtonOwnProps extends CoreButtonProps {
+  /**Skin of the Button (Styling)*/
+  skin?: Skin;
+  /** Priority of the Button (Styling)*/
+  priority?: Priority;
+  /** Size of the button (Styling) */
+  size?: Size;
+}
+
+export type ButtonProps = ButtonOwnProps & CoreButtonProps;
+
+export const Button : React.SFC<ButtonProps> = props => {
+  const {children, skin, priority, size, ...rest} = props;
+
+  return (
+    <CoreButton 
+      {...rest}
+      {...style('root',{skin, priority, size}, rest)}
+    >
+      {children}
```

Comment:
I would prefer you add the Text in this pr, because it is something that can be forgotten, unlike error state or hover state...

Reply:
It looks so ugly I wont forget.